Sources/Playlist/PlaylistController.swift

```diff
@@ -0,0 +1,137 @@
+    
+    /// Start playlist from index 0
+    func replay()
+    
+    /// Set this parameter to true if you need to play the list from the very beginning after last item ends.
+    /// Default value is false.
+    var loop: Bool { get set }
+    
+    /// Continue to the next media item after current one ends.
+    /// Default value is true.
+    var autoContinue: Bool { get set }
+    
+    /// Play next item if current cannot be loaded or any playback error occurred.
+    /// Default value is true.
+    var recoverOnError: Bool { get set }
+    
+    /// Index of currently playing media.
+    var currentMediaIndex: Int { get }
+    
+    /// Disables coundown action for current media
+    func disableCountdownForCurrentItem()
+    
+    /// Checks if the previous item in the list is available for playback.
+    /// If loop is true isPreviousItemAvailable() will always return true.
+    func isPreviousItemAvailable() -> Bool
+    
+    /// Checks if the next item in the list is available for playback.
+    /// If loop is true isNextItemAvailable() will always return true.
+    func isNextItemAvailable() -> Bool
+}
+
+@objc public protocol PlaylistControllerDelegate: AnyObject {
+    
+    /**
+     Individual media items can opt out of having the dedicated plugin config for each media item.
+     
+     * Parameters:
+        * controller: The PlaylistController which is managing current medias list playback.
+        * mediaItemIndex: The index of media item.
+     * Returns: A flag if it is needed to set dedicated plugins config to item with requested index.
+     */
+    func playlistController(_ controller: PlaylistController, updatePluginConfigForMediaItemAtIndex mediaItemIndex: Int) -> Bool
+    
+    /**
+     Individual media items can opt out of having the dedicated plugin config for each media item.
+     This method will be called only if playlistController(: updatePluginConfigForMediaItemAtIndex:) returns true.
+     
+     * Parameters:
+        * controller: The PlaylistController which is managing current medias list playback.
+        * mediaItemIndex: The index of media item.
+     * Returns: A plugins config which will be applied to item with requested index.
+     In case it returns nil default plugins config will be applied for the media at requested index.
+     */
+    func playlistController(_ controller: PlaylistController, pluginConfigForMediaItemAtIndex mediaItemIndex: Int) -> PluginConfig?
+    
+    /**
+     Individual media items can opt out of having the countdown options applied to their playback.
+     
+     * Parameters:
+        * controller: The PlaylistController which is managing current medias list playback.
+        * mediaItemIndex: The index of media item.
+     * Returns: A flag if it is needed to apply countdown options to item with requested index.
+     */
+    func playlistController(_ controller: PlaylistController, enableCountdownForMediaItemAtIndex mediaItemIndex: Int) -> Bool
```

Comment:
After adding the playlist to the OVP and OTT samples I saw that the MediaEntry itself is needed in order to not show the countdown view for live media. I will commit the changes afterwards.